src/test/java/org/xstefank/model/PersistentListTest.java

```diff
@@ -3,55 +3,162 @@
 import java.io.File;
 import java.io.FileWriter;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.BeforeClass;
 import org.junit.Test;
+import org.xstefank.FileLine;
 import org.xstefank.TestUtils;
 
 public class PersistentListTest {
 
-    private static final String FILE_NAME = "testList.txt";
-    private static File userListFile = new File(TestUtils.TARGET_DIR, FILE_NAME);
+    private static final String FILE_NAME = "list.txt";
+    private static File persistentListFile = new File(TestUtils.TARGET_DIR, FILE_NAME);
 
-    private final String USERNAME = "testUser";
-    private PersistentList testUserList;
+    private final String FIRST = "first";
+    private final String SECOND = "second";
+
+    private PersistentList persistentList;
 
     @BeforeClass
     public static void beforeClass() {
-        TestUtils.deleteFileIfExists(userListFile);
+        TestUtils.deleteFileIfExists(persistentListFile);
     }
 
     @Before
     public void before() {
-        testUserList = new PersistentList(userListFile);
-    }
-
-    @Test
-    public void testIfUserIsSaved() {
-        testUserList.addUser(USERNAME);
-        Assert.assertTrue(testUserList.hasUsername(USERNAME));
-        Assert.assertTrue(TestUtils.fileContainsLine(userListFile, USERNAME));
+        persistentList = new PersistentList(persistentListFile);
     }
 
     @Test
     public void testIfFileWasCreated() {
-        Assert.assertTrue(userListFile.exists());
+        Assert.assertTrue(persistentListFile.exists());
     }
 
     @Test
     public void testFileContentLoading() throws IOException {
-        FileWriter fw = new FileWriter(userListFile);
-        fw.write(USERNAME);
+        FileWriter fw = new FileWriter(persistentListFile);
+        fw.write(FIRST);
         fw.close();
 
-        testUserList = new PersistentList(userListFile);
-        Assert.assertTrue(testUserList.hasUsername(USERNAME));
+        persistentList = new PersistentList(persistentListFile);
+        Assert.assertTrue(persistentList.contains(FIRST));
+    }
+
+    @Test
+    public void testSetElement() {
+        persistentList.add(FIRST);
+        int index = persistentList.indexOf(FIRST);
+        persistentList.set(index, SECOND);
+        FileLine line = TestUtils.getSpecificLine(persistentListFile, SECOND);
+
+        Assert.assertTrue(persistentList.contains(SECOND));
+        Assert.assertSame(index, persistentList.indexOf(SECOND));
+        Assert.assertSame(index + 1, line.getLineNumber());
```

Comment:
be careful with assertSame vs assertEquals

Reply:
I though for numbers it is OK to use assertSame

Reply:
yes, they are, but be careful with for instance int == int and Integer == Integer